### PR TITLE
fix(core): OAuth1 authentication fix for Clever Cloud API

### PR DESCRIPTION
**Pull request by michael-radency**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

**Review by netroy — DISMISSED** [review]
we should eventually abstract away all of this `clientOAuth1` code like we are now doing for oauth2. but for now, this code LGTM 👍🏽 🙏🏽

**Comment by michael-radency** [comment]
> we should eventually abstract away all of this `clientOAuth1` code like we are now doing for oauth2. but for now, this code LGTM 👍🏽 🙏🏽

unfortunately this is still WIP, as authentication to Clever Cloud continue to fail with 415

File: packages/cli/src/Server.ts [hunk 1]
```diff
@@ -923,7 +923,19 @@ export class Server extends AbstractServer {
...
+						let algorithm: string;
+						switch (signatureMethod) {
+							case 'HMAC-SHA256':
+								algorithm = 'sha256';
+								break;
+							case 'HMAC-SHA512':
+								algorithm = 'sha512';
+								break;
+							default:
+								algorithm = 'sha1';
```
Comment (netroy) [inline packages/cli/src/Server.ts:935]: do you know if we really have nodes that use/support `sha1`? 
[NIST deprecated `sha1` 12 years ago](https://csrc.nist.gov/pubs/sp/800/131/a/r1/final), strongly discourages any software that uses `sha1`, and [will block purchase of all such software by government agencies after 2030](https://www.nist.gov/news-events/news/2022/12/nist-retires-sha-1-cryptographic-algorithm).

Comment (michael-radency) [reply packages/cli/src/Server.ts:935]: no, I do not have such info,
OAuth1 rarely used in nodes, currently only in TwitterV1, GraphQL and HttpRequest

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.2.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.2.1)

**Pull request merged**